internal/pke/workflow/assemble_http_proxy_settings_activity_test.go

```diff
@@ -0,0 +1,161 @@
+// limitations under the License.
+
+package workflow
+
+import (
+	"context"
+	"testing"
+
+	"emperror.dev/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAssembleHTTPProxySettings(t *testing.T) {
+	orgID := uint(13)
+	httpSecretID := "http-secret-id"
+	httpsSecretID := "https-secret-id"
+
+	a := MakeAssembleHTTPProxySettingsActivity(dummyPasswordSecretStore{
+		entries: map[uint]map[string]dummyPasswordSecret{
+			orgID: {
+				httpSecretID: {
+					username: "http-username",
+					password: "http-password",
+				},
+				httpsSecretID: {
+					username: "https-username",
+					password: "https-password",
+				},
+			},
+		},
+	})
+
+	tc := map[string]struct {
+		Input  AssembleHTTPProxySettingsActivityInput
+		Output HTTPProxy
+		Error  interface{}
+	}{
+		"with everything": {
+			Input: AssembleHTTPProxySettingsActivityInput{
+				OrganizationID:     orgID,
+				HTTPProxyHostPort:  "http-proxy.org:1234",
+				HTTPProxySecretID:  httpSecretID,
+				HTTPSProxyHostPort: "https-proxy.org:2345",
+				HTTPSProxySecretID: httpsSecretID,
+			},
+			Output: HTTPProxy{
+				HTTPProxyURL:  "http://http-username:http-password@http-proxy.org:1234",
+				HTTPSProxyURL: "https://https-username:https-password@https-proxy.org:2345",
+			},
+		},
+		"http only with secret": {
+			Input: AssembleHTTPProxySettingsActivityInput{
+				OrganizationID:    orgID,
+				HTTPProxyHostPort: "http-proxy.org:1234",
+				HTTPProxySecretID: httpSecretID,
+			},
+			Output: HTTPProxy{
+				HTTPProxyURL: "http://http-username:http-password@http-proxy.org:1234",
+			},
+		},
+		"http only without secret": {
+			Input: AssembleHTTPProxySettingsActivityInput{
+				OrganizationID:    orgID,
+				HTTPProxyHostPort: "http-proxy.org:1234",
+			},
+			Output: HTTPProxy{
+				HTTPProxyURL: "http://http-proxy.org:1234",
+			},
+		},
+		"https only with secret": {
+			Input: AssembleHTTPProxySettingsActivityInput{
+				OrganizationID:     orgID,
+				HTTPSProxyHostPort: "https-proxy.org:2345",
+				HTTPSProxySecretID: httpsSecretID,
+			},
+			Output: HTTPProxy{
+				HTTPSProxyURL: "https://https-username:https-password@https-proxy.org:2345",
+			},
+		},
+		"https only without secret": {
+			Input: AssembleHTTPProxySettingsActivityInput{
+				OrganizationID:     orgID,
+				HTTPSProxyHostPort: "https-proxy.org:2345",
+			},
+			Output: HTTPProxy{
+				HTTPSProxyURL: "https://https-proxy.org:2345",
+			},
+		},
+		"empty": {
+			Input:  AssembleHTTPProxySettingsActivityInput{},
+			Output: HTTPProxy{},
+		},
+		"missing secret": {
+			Input: AssembleHTTPProxySettingsActivityInput{
+				OrganizationID:     uint(24),
+				HTTPProxyHostPort:  "http-proxy.org:1234",
+				HTTPProxySecretID:  httpSecretID,
+				HTTPSProxyHostPort: "https-proxy.org:2345",
+				HTTPSProxySecretID: httpsSecretID,
+			},
+			Error: true,
+		},
+	}
+	for name, tv := range tc {
+		t.Run(name, func(t *testing.T) {
+			output, err := a.Execute(context.Background(), tv.Input)
+			switch tv.Error {
```

Comment:
It's usually better to have error scenarios in a separate test case.

Reply:
Yes, that's why there are sub-tests per test case for every scenario.

Reply:
I meant have one (with subtests) for the happy path and another (with potential subtests) for the error path. But it doesn't matter that much.